featurebyte/sql/databricks_unity/F_COUNT_DICT_COSINE_SIMILARITY.sql

```diff
@@ -29,7 +29,9 @@ $$
   for k, v in counts_other.items():
     v = v or 0
     norm_other += v * v
-  cosine_sim = dot_product / (math.sqrt(norm) * math.sqrt(norm_other))
-  return cosine_sim
+  norm_product = math.sqrt(norm) * math.sqrt(norm_other)
+  if norm_product == 0:
+    return 0
```

Comment:
Should we return `null` instead of 0 here?

Reply:
Or we should return 1 instead of 0:
```
In [2]: from scipy.spatial import distance

...
In [7]: 1 - distance.cosine([1, 0], [0, 0])
/Users/senbonggee/Library/Caches/pypoetry/virtualenvs/featurebyte-Ryj2vg6L-py3.8/lib/python3.8/site-packages/scipy/spatial/distance.py:622: RuntimeWarning: invalid value encountered in scalar divide
  dist = 1.0 - uv / np.sqrt(uu * vv)
Out[7]: 1
```

Reply:
If i understood it correctly, `1 - distance.cosine(u, v)` defines cosine similarity (https://stackoverflow.com/questions/18424228/cosine-similarity-between-2-number-lists).

Reply:
Maybe we should not use the cosine similar & distance formula when one of the vector has 0 norm. Let's keep it to 0 as it makes more sense.

Reply:
the existing behavior is to return 0 for 2 empty dict so 0 will be more consistent